crates/typst-macros/src/elem.rs

```diff
@@ -1301,3 +1333,30 @@ fn create_field_parser(field: &Field) -> (TokenStream, TokenStream) {
 
     (quote! {}, value)
 }
+
+pub fn is_option(path: &syn::Type) -> bool {
+    let opt = match path {
+        syn::Type::Path(typepath) if typepath.qself.is_none() => {
+            Some(typepath.path.clone())
+        }
+        _ => None,
+    };
+
+    if let Some(o) = opt {
+        check_for_option(&o).is_some()
+    } else {
+        false
+    }
+}
+
+fn check_for_option(path: &syn::Path) -> Option<&syn::PathSegment> {
+    let idents_of_path = path.segments.iter().fold(String::new(), |mut acc, v| {
+        acc.push_str(&v.ident.to_string());
+        acc.push(':');
+        acc
+    });
+    ["Option:", "std:option:Option:", "core:option:Option:"]
+        .into_iter()
+        .find(|s| idents_of_path == *s)
+        .and_then(|_| path.segments.last())
+}
```

Comment:
This check seems overcomplicated, just a check for `Option` would suffice I think. But even if not, I don't think allocation is necessary.